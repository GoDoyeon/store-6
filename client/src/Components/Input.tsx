import styled from "styled-components";
import underlineSVG from '~/assets/underline.svg';

const Input = styled.input`
  font-size: 1.2rem;
  margin-top: 1rem;
  width: 100%;
  padding-left: 2rem;
  height: 2rem;
  background: url(${underlineSVG}) bottom left no-repeat;
  box-sizing: border-box;
  background-size: contain;
`;

export default Input;
