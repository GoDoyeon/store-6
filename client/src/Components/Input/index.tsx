import styled from 'styled-components';
import { underlineSVG } from '~/assets/index';

/**
 * width에 string으로 %를 넘겨주세요.
 *
 * @example
 * <Input width="100%" />
 */
const Input = styled.input<{ width?: string }>`
  box-sizing: border-box;
  width: ${(props) => props.width || '100%'};
  height: 34px;
  padding: 0 16px 10px 16px;
  margin-top: 16px;
  line-height: 25px;
  font-size: 20px;
  background: bottom left / contain no-repeat url(${underlineSVG});

  &::placeholder {
    color: #999;
  }
`;

export default Input;
