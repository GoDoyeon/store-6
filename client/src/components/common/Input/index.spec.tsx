import { render } from '@testing-library/react';
import 'jest-styled-components';

import Input from '~/components/common/Input';

describe('<Input />', () => {
  it('should render same with snapshot', () => {
    const { container } = render(<Input />);
    expect(container).toMatchSnapshot();
  });
});
