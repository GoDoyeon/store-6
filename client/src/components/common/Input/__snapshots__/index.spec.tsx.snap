// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render same with snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  height: 34px;
  padding: 0 16px 10px 16px;
  margin-top: 16px;
  line-height: 25px;
  font-size: 20px;
  background: bottom left / contain no-repeat url();
}

.c0::-webkit-input-placeholder {
  color: #999;
}

.c0::-moz-placeholder {
  color: #999;
}

.c0:-ms-input-placeholder {
  color: #999;
}

.c0::placeholder {
  color: #999;
}

<div>
  <input
    class="c0"
  />
</div>
`;
